ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx

```diff
@@ -263,6 +273,19 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
         this.setState(newChanges);
     }
 
+    private getNode(nodeId: string): NodeStatus {
+        const node: NodeStatus = this.getNode(nodeId);
+        if (!node) {
+            // We may arrive here if we retry a workflow with a failed pod. During the retry operation, the node corresponding
+            // for the failed pod gets deleted, but the child reference of the parent nodes does not, causing them to refer to
+            // a node that doesn't exist. As soon as the workflow gets operated on again, the node will be recreated. For the
+            // time before that happens, don't error out.
```

Comment:
Another approach to this fix would be to delete the child references in parent nodes when retrying a Workflow... I think I may favor that approach instead. Thoughts?

Reply:
better to fix the underlying data

Reply:
Will do both, I think it's best to leave this in the UI as a stop-gap

Reply:
I added the fix for the underlying data. I should probably remove this comment then?

Reply:
sure!